app/uk/gov/hmrc/helptosavestub/controllers/NSIController.scala

```diff
@@ -107,6 +112,75 @@ object NSIController extends BaseController with Logging {
 
   private val ninoStatusRegex = """AS(\d{3}).*""".r
 
+  def getAccount(correlationId: Option[String], nino: Option[String], version: Option[String], systemId: Option[String]): Action[AnyContent] = Action { implicit request ⇒
+    validateParams(correlationId, nino, version, systemId)
+      .fold(
+        errors ⇒ {
+          logger.warn("[Handle Account Query] invalid params")
+          errors match {
+            case error :: Nil ⇒ BadRequest(Json.toJson(error))
+            case _            ⇒ BadRequest
+          }
+        }, {
+          case nino ⇒
+            if (nino.contains("401")) {
+              Unauthorized
+            } else if (nino.contains("500")) {
+              InternalServerError
+            } else {
+              val maybeAccount = getAccountByNino(nino, correlationId)
+              maybeAccount match {
+                case Right(a) ⇒ Ok(Json.toJson(a))
+                case Left(e)  ⇒ BadRequest(Json.toJson(e))
+              }
+            }
+          case other ⇒
```

Comment:
There are a few Scala compiler warnings on this file, some of them indicate that some of the new code has no effect, please can you investigate:
```
Warning:(137, 16) [IdenticalCaseConditions] Identical case condition detected above. This case will never match.
          case other ⇒
Warning:(167, 12) [IdenticalCaseConditions] Identical case condition detected above. This case will never match.
      case other ⇒ Invalid(NSIErrorResponse.badNinoResponse(correlationId))
Warning:(125, 16) patterns after a variable pattern cannot match (SLS 8.1.1)
If you intended to match against parameter nino of method getAccount, you must use backticks, like: case `nino` =>
          case nino ⇒
Warning:(138, 13) unreachable code due to variable pattern 'nino' on line 125
            BadRequest
Warning:(138, 13) unreachable code
            BadRequest
Warning:(166, 12) patterns after a variable pattern cannot match (SLS 8.1.1)
If you intended to match against parameter nino of method validateParams, you must use backticks, like: case `nino` =>
      case nino  ⇒ if (!nino.matches(helptosavestub.util.ninoRegex.regex)) Invalid(NSIErrorResponse.badNinoResponse(correlationId)) else Valid(nino)
Warning:(167, 27) unreachable code due to variable pattern 'nino' on line 166
      case other ⇒ Invalid(NSIErrorResponse.badNinoResponse(correlationId))
Warning:(167, 27) unreachable code
      case other ⇒ Invalid(NSIErrorResponse.badNinoResponse(correlationId))
Warning:(148, 15) [UnusedParameter] Parameter systemId is not used in method validateParams.
  private def validateParams(correlationId: Option[String], nino: Option[String], version: Option[String], systemId: Option[String]): Either[List[NSIErrorResponse], NINO] = {
```